source/communication_interfaces/test/tests/test_udp_communication.cpp

```diff
@@ -0,0 +1,82 @@
+#include <gtest/gtest.h>
+
+#include "communication_interfaces/exceptions/SocketConfigurationException.hpp"
+#include "communication_interfaces/sockets/UDPClient.hpp"
+#include "communication_interfaces/sockets/UDPServer.hpp"
+
+using namespace communication_interfaces;
+
+class TestUDPSockets : public ::testing::Test {
+public:
+  TestUDPSockets() {
+    params_.emplace_back(state_representation::make_shared_parameter<std::string>("ip_address", "127.0.0.1"));
+    params_.emplace_back(state_representation::make_shared_parameter<int>("port", 5000));
+    params_.emplace_back(state_representation::make_shared_parameter<int>("buffer_size", 100));
+  }
+
+  std::list<std::shared_ptr<state_representation::ParameterInterface>> params_;
+};
+
+TEST_F(TestUDPSockets, SendReceive) {
+  const std::string send_string = "Hello world!";
+  params_.emplace_back(state_representation::make_shared_parameter<int>("buffer_size", 100));
```

Comment:
this line seems redundant because you already set up the params property in the fixture with this value. If you have duplicates in the list it will just re-validate and set the parameter I believe.